crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -310,8 +310,18 @@ impl Settlement {
             .fold(Default::default(), |mut acc, solution| {
                 for trade in solution.user_trades() {
                     let order = acc.entry(trade.order().uid).or_default();
-                    order.sell = trade.sell_amount(&solution.prices).unwrap_or_default();
-                    order.buy = trade.buy_amount(&solution.prices).unwrap_or_default();
+                    order.sell = trade.sell_amount(&solution.prices, solution.weth).unwrap_or_else(|| {
+                        // This should never happen, returning 0 is better than panicking, but we
+                        // should still alert.
+                        tracing::error!(uid = ?trade.order().uid, "could not compute sell_amount");
```

Comment:
Probably makes sense to log the whole `trade` and `clearing_prices` because we otherwise don't really know exactly why things failed.
Also this is currently an alert that can be triggered directly by a solver reporting a weird solution. I think they probably won't actively try to annoy us with that but we might have to silence it in the future (or demote to `warn`).

Reply:
Honestly I'm more worried about us having a logic error in the way we compute sell amounts (which we already had cf. https://github.com/cowprotocol/services/pull/2148) so I'd make it alert by default for now.

Reply:
The alert was not my main point here. I was mostly concerned with logging more data so we can easily figure out what part of the calculation failed. If we only log the `uid` it would be harder to gather all the relevant data to recreate the computation.